src/cls/IPM/ResourceProcessor/WebApplication.cls

```diff
@@ -0,0 +1,230 @@
+}
+
+/// Keeps track of all application properties
+Property SecurityProperties As %RawString [ MultiDimensional ];
+
+Method CopyAttributes() [ Private ]
+{
+    /*
+     * Overridden to avoid needing a property in this class to correspond to each
+     * attribute as the attribute list is just all non-private/transient properties 
+     * in Security.Applications.
+     */
+    Set key = ""
+    While 1 {
+        Set value = ..ResourceReference.Attributes.GetNext(.key)
+        If (key = "") {
+            Quit
+        }
+        Set eval = ..%Evaluate(value)
+        Set ..SecurityProperties(key) = eval
+    }
+}
+
+/// Called as phase <var>pPhase</var> is executed for the resource. If <var>pResourceHandled</var> is set to true,
+/// then the default behavior for that resource will be bypassed in the current phase.
+Method OnPhase(pPhase As %String, ByRef pParams, Output pResourceHandled As %Boolean = 0) As %Status
+{
+    Set sc = $$$OK
+    Try {
+        If (pPhase = "Validate") {
+            If (..Name = "") {
+                Return $$$ERROR($$$GeneralError,"Name is required for web applications")
+            }
+            Set keyList = ""
+            Set key = ""
+            While 1 {
+                Set value = ..ResourceReference.Attributes.GetNext(.key)
+                If (key = "") {
+                    Quit
+                }
+                Set keyList = keyList _ $ListBuild(key)
+            }
+            Set badKeyList = ..DoPropertiesExist(keyList)
+            If ($ListLength(badKeyList) > 0) {
+                Set error = "The following invalid attributes were found for resource "_..ResourceReference.Name_": "_$ListToString(badKeyList)
+                $$$ThrowStatus($$$ERROR($$$GeneralError,error))
+            }
+        }
+    } Catch (ex) {
+        Set sc = ex.AsStatus()
+    }
+    Return sc
+}
+
+Method OnBeforePhase(pPhase As %String, ByRef pParams) As %Status
+{
+    /*
+     * NOTE: Web app creation is done in OnBeforePhase so it happens before other
+     * resource processors such as FileCopy run.
+     */
+    Set sc = $$$OK
+    Try {
+        $$$ThrowOnError(##super(pPhase,.pParams))
+        
+        Set tVerbose = $Get(pParams("Verbose"))
+        
+        If ((pPhase = "Configure") || (pPhase = "Activate")) {
+            // Create web application
+            Do ..CreateOrUpdateWebApp(tVerbose)
+        } ElseIf ((pPhase = "Unconfigure") || (pPhase = "Clean")) {
+            // Remove web application
+            Do ..DeleteWebApp(tVerbose)
+        }
+    } Catch e {
+        Set sc = e.AsStatus()
+    }
+    Quit sc
+}
+
+Method CreateOrUpdateWebApp(pVerbose As %Boolean = 0) [ Internal ]
+{
+    // TODO: Make mirror-safe.
+    New $namespace
+    Set $namespace = "%SYS"
+
+    Merge properties = ..SecurityProperties
+    If $Get(properties("AutheEnabled")) = "" {
+        Set properties("AutheEnabled") = (
+            ($Get(properties("PasswordAuthEnabled")) * $$$AutheCache) +
```

Comment:
I don't think we should have these sort of flags at all since it will go stale if a new auth type is introduced (for example, oauth will be a new type in IRIS 2025.1). Best to just pass through to Security.Applications APIs but allow for expressions to make it less abstract/simpler instead of having the numbers directly.

Reply:
I am okay with using comma-separated indices.

Reply:
@isc-shuliu I agree with @isc-kiyer that we should add expression support for this. To be more specific, this might look like:
${authPassword} -> $$$AutheCache
${authUnauthenticated} -> $$$AutheUnauthenticated
${authDelegated} -> $$$AutheDelegated
etc.

That works well if you support a single mode; not sure if this would work, but would be interesting to see if it does (and perhaps we should make it so):
`AutheEnabled="#{${authDelegated} + ${authPassword}}"`

If a new mode is released that IPM doesn't yet support, a user could add the relevant integer without the expression and add a comment about it.

Reply:
@isc-tleavitt How does this commit https://github.com/intersystems/ipm/pull/562/commits/1c3892153759780fde94b62b64724e6a327a3a51 look to you? 

Basically we substitute ${authPassword} in the same way we handle ${namespace}, ${installDir}, etc., and separate the string by "+", and finally add the numbers up. 

Comments are also made possible because "32 abc" is treated as 32 during addition in ObjectScript